### ISSUE TITLE
[zod/open-api] Problem with RPC types

### ISSUE THREAD

**Issue by OxYz0n3** [honojs/middleware#513]
There is few problems with RPC Mode of zod-openapi middleware.

1. When creating a route, if no 'request' field is specified (like for a GET route), the route is wrongly typed when using typeof router and it screw up auto completion and return types for all of the client.

2. Even if we set responses in the route, the types of the responses will be retrieved, but not the status code so it can't be used.

![image](https://github.com/honojs/middleware/assets/71818947/204501b6-6713-4254-9c9e-ebf3e7c2ab5f)

Example code:
```ts
import { createRoute, OpenAPIHono, z } from "@hono/zod-openapi";
import { hc } from "hono/client";

const router = new OpenAPIHono()
    .openapi(createRoute({
        method: 'get',
        path: '/incorrect',
        request: { // Comment this field and every types will be screwd up
            params: z.object({
                param: z.string()
            })
        },
        responses: {
            200: {
                description: 'Test',
                content: {
                    'application/json': {
                        schema: z.object({
                            ok: z.boolean()
                        })
                    }
                }
            }, 400: {
                description: 'error',
                content: {
                    'application/json': {
                        schema: z.object({
                            msg: z.string()
                        })
                    }
                }
            }
        }
    }), async c => {
        if (c.req.param('param') === 'test')
            return (c.json({ ok: true }, 200));
        else
            return c.json({ msg: 'Error!' }, 400);
    })
    .get('/correct', async c => {
        if (c.req.param('param') === 'test')
            return (c.json({ ok: true }, 200));
        else
            return c.json({ msg: 'Error!' }, 400);
    });

type routerType = typeof router;

const client = hc<routerType>("http://localhost:3000");

client.correct.$get({ param: { param: "test" }}); // Works good

client.incorrect.$get({ param: { param: "test" }}); // Look at return types, and then try to comment the request field in the route

```
hono version: 4.3.4
@hono/zod-openapi: 0.11.1

**Comment by lukeshay** [comment]
I am running into this as well.

**Comment by uyrabi** [comment]
It might not be the same case, but I am also encountering issues with OpenAPI and RPC.

[versions]
hono: 4.3.6
@hono/zod-openapi@0.11.1

Here is the client:

```
const client = hc<typeof hoge>('http://localhost:3000/api');
```

Up to hc<typeof hoge>, the type is active:

```
(alias) hc<OpenAPIHono<Env, {
    "/hoge": {
        $post: {
            input: {
                param: {
                    date: string;
                    id: number;
                };
            };
```

However, the client is not functioning with types:

```
const client: {
    [x: string]: ClientRequest<{
        [Method: `$${Lowercase<string>}`]: Endpoint;
    }>;
}
```

**Comment by yusukebe** [comment]
These issues may be fixed with #519, and I released the new version that includes that fix.

Notice. From the version, you need to specify the status code in `c.json()` even if it is `200`:

```ts
app.openapi(route, (c) => {
  const { id } = c.req.valid('param')
  return c.json(
    {
      id,
      age: 20,
      name: 'Ultra-man',
    },
    200 // You should specify the status code even if it is 200.
  )
})
```
